backend/packages/Upgrade/src/api/services/ExperimentUserService.ts

```diff
@@ -298,14 +298,22 @@ export class ExperimentUserService {
     return this.userRepository.save(newDocument);
   }
 
-  public async getUserDoc(experimentUserId, logger): Promise<RequestedExperimentUser> {
+  public async getUserDoc(
+    experimentUserId: string,
+    logger: UpgradeLogger,
+    api?: string
```

Comment:
This is not a good way. The function is doing more than what the name is suggesting.